linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpIdentifierConfig.scala

```diff
@@ -1,12 +1,16 @@
 package io.buoyant.linkerd.protocol
 
-import com.fasterxml.jackson.annotation.{JsonIgnore, JsonTypeInfo}
+import com.fasterxml.jackson.annotation.{JsonIgnore, JsonProperty, JsonTypeInfo}
 import com.twitter.finagle.http.Request
 import com.twitter.finagle.{Dtab, Path}
 import io.buoyant.router.RoutingFactory.Identifier
 
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "kind")
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "kind", visible = true)
 trait HttpIdentifierConfig {
+
+  @JsonProperty("kind")
```

Comment:
TIOLI: could create a `Config` trait that defines this var and have each Config class extend it.  That would reduce the boilerplate a bit.